gdk4/src/toplevel_size.rs

```diff
@@ -0,0 +1,144 @@
+        unsafe {
+            ffi::gdk_toplevel_size_set_min_size(self.to_glib_none_mut().0, min_width, min_height);
+        }
+    }
+
+    #[doc(alias = "gdk_toplevel_size_set_shadow_width")]
+    pub fn set_shadow_width(&mut self, left: i32, right: i32, top: i32, bottom: i32) {
+        unsafe {
+            ffi::gdk_toplevel_size_set_shadow_width(
+                self.to_glib_none_mut().0,
+                left,
+                right,
+                top,
+                bottom,
+            );
+        }
+    }
+
+    #[doc(alias = "gdk_toplevel_size_set_size")]
+    pub fn set_size(&mut self, width: i32, height: i32) {
+        unsafe {
+            ffi::gdk_toplevel_size_set_size(self.to_glib_none_mut().0, width, height);
+        }
+    }
+}
+
+#[doc(hidden)]
+impl<'a> ToGlibPtr<'a, *const ffi::GdkToplevelSize> for ToplevelSize {
+    type Storage = &'a Self;
+
+    #[inline]
+    fn to_glib_none(&'a self) -> Stash<'a, *const ffi::GdkToplevelSize, Self> {
+        let ptr: *const ToplevelSize = &*self;
+        Stash(ptr as *const ffi::GdkToplevelSize, self)
+    }
+}
+
+#[doc(hidden)]
+impl<'a> ToGlibPtrMut<'a, *mut ffi::GdkToplevelSize> for ToplevelSize {
+    type Storage = &'a mut Self;
+
+    #[inline]
+    fn to_glib_none_mut(&'a mut self) -> StashMut<'a, *mut ffi::GdkToplevelSize, Self> {
+        let ptr: *mut ToplevelSize = &mut *self;
+        StashMut(ptr as *mut ffi::GdkToplevelSize, self)
+    }
+}
+
+#[doc(hidden)]
+impl FromGlibPtrNone<*const ffi::GdkToplevelSize> for ToplevelSize {
+    unsafe fn from_glib_none(ptr: *const ffi::GdkToplevelSize) -> Self {
+        *(ptr as *const ToplevelSize)
+    }
+}
+
+#[doc(hidden)]
+impl FromGlibPtrNone<*mut ffi::GdkToplevelSize> for ToplevelSize {
+    unsafe fn from_glib_none(ptr: *mut ffi::GdkToplevelSize) -> Self {
+        *(ptr as *mut ToplevelSize)
+    }
+}
+
+#[doc(hidden)]
+impl FromGlibPtrBorrow<*const ffi::GdkToplevelSize> for ToplevelSize {
+    unsafe fn from_glib_borrow(
+        ptr: *const ffi::GdkToplevelSize,
+    ) -> glib::translate::Borrowed<Self> {
+        glib::translate::Borrowed::new(*(ptr as *const ToplevelSize))
+    }
+}
+
+#[doc(hidden)]
+impl FromGlibPtrBorrow<*mut ffi::GdkToplevelSize> for ToplevelSize {
+    unsafe fn from_glib_borrow(ptr: *mut ffi::GdkToplevelSize) -> glib::translate::Borrowed<Self> {
+        glib::translate::Borrowed::new(*(ptr as *mut ToplevelSize))
+    }
+}
+
+#[doc(hidden)]
+impl FromGlibPtrFull<*mut ffi::GdkToplevelSize> for ToplevelSize {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *mut ffi::GdkToplevelSize) -> Self {
+        *(ptr as *mut ToplevelSize)
+    }
+}
+
+impl glib::StaticType for ToplevelSize {
+    #[doc(alias = "gdk_toplevel_size_get_type")]
+    fn static_type() -> glib::types::Type {
+        skip_assert_initialized!();
+        unsafe { from_glib(ffi::gdk_toplevel_size_get_type()) }
+    }
+}
+
+impl<'a> glib::value::FromValueOptional<'a> for ToplevelSize {
```

Comment:
Does it make sense to have `StaticType` and all the `GValue` machinery? I don't think so